File: src/components/JSONHandler/src/ALRPCObjectsImpl/SoftButton.cpp
#include "../include/JSONHandler/ALRPCObjects/SoftButton.h"
#include "SoftButtonMarshaller.h"
#include "ImageMarshaller.h"
#include "SoftButtonTypeMarshaller.h"
#include "SystemActionMarshaller.h"
/*
  interface	Ford Sync RAPI
  version	1.2 / 2.0O
  date		2011-05-17 / 2012-11-02
  generated at	Thu Nov 29 14:49:08 2012
  source stamp	Thu Nov 29 06:50:10 2012
  author	robok0der
*/



using namespace NsAppLinkRPC;

SoftButton::SoftButton(const SoftButton& c)
{
  *this=c;
}


bool SoftButton::checkIntegrity(void)
{
  return SoftButtonMarshaller::checkIntegrity(*this);
}


SoftButton::SoftButton(void)
{
}



bool SoftButton::set_image(const Image& image_)
{
  if(!ImageMarshaller::checkIntegrityConst(image_))   return false;
  image=image_;
  return true;
}

bool SoftButton::set_isHighlighted(bool isHighlighted_)
{
  isHighlighted=isHighlighted_;
  return true;
}

bool SoftButton::set_softButtonID(unsigned int softButtonID_)
{
  if(softButtonID_>65536)  return false;
  softButtonID=softButtonID_;
  return true;
}

bool SoftButton::set_systemAction(const SystemAction& systemAction_)
{
  if(!SystemActionMarshaller::checkIntegrityConst(systemAction_))   return false;
  systemAction=systemAction_;
  return true;
}

bool SoftButton::set_text(const std::string& text_)
{
  if(text_.length()>500)  return false;
  text=text_;
  return true;
}

bool SoftButton::set_type(const SoftButtonType& type_)
{
  if(!SoftButtonTypeMarshaller::checkIntegrityConst(type_))   return false;
  type=type_;
  return true;
}




const Image& SoftButton::get_image(void) const 
{
  return image;
}


bool SoftButton::get_isHighlighted(void) const
{
  return isHighlighted;
}


unsigned int SoftButton::get_softButtonID(void) const
{
  return softButtonID;
}


const SystemAction& SoftButton::get_systemAction(void) const 
{
  return systemAction;
}


const std::string& SoftButton::get_text(void) const 
{
  return text;
}


const SoftButtonType& SoftButton::get_type(void) const 
{
  return type;
}

File: mobile/android/SyncProxyAndroid/src/main/java/com/ford/syncV4/proxy/rpc/SoftButton.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCStruct;
import com.ford.syncV4.proxy.constants.Names;
import com.ford.syncV4.proxy.rpc.enums.SoftButtonType;
import com.ford.syncV4.proxy.rpc.enums.SystemAction;
import com.ford.syncV4.util.DebugTool;

public class SoftButton extends RPCStruct {

    public SoftButton() { }
    public SoftButton(Hashtable hash) {
        super(hash);
    }
    public void setType(SoftButtonType type) {
        if (type != null) {
            store.put(Names.type, type);
        } else {
        	store.remove(Names.type);
        }
    }
    public SoftButtonType getType() {
    	Object obj = store.get(Names.type);
        if (obj instanceof SoftButtonType) {
            return (SoftButtonType) obj;
        } else if (obj instanceof String) {
        	SoftButtonType theCode = null;
            try {
                theCode = SoftButtonType.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.type, e);
            }
            return theCode;
        }
        return null;
    }
    public void setText(String text) {
        if (text != null) {
            store.put(Names.text, text);
        } else {
        	store.remove(Names.text);
        }
    }
    public String getText() {
        return (String) store.get(Names.text);
    }
    public void setImage(Image image) {
        if (image != null) {
            store.put(Names.image, image);
        } else {
        	store.remove(Names.image);
        }
    }
    public Image getImage() {
    	Object obj = store.get(Names.image);
        if (obj instanceof Image) {
            return (Image) obj;
        } else if (obj instanceof Hashtable) {
        	try {
        		return new Image((Hashtable) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.image, e);
            }
        }
        return null;
    }
    public void setIsHighlighted(Boolean isHighlighted) {
        if (isHighlighted != null) {
            store.put(Names.isHighlighted, isHighlighted);
        } else {
        	store.remove(Names.isHighlighted);
        }
    }
    public Boolean getIsHighlighted() {
        return (Boolean) store.get(Names.isHighlighted);
    }
    public void setSoftButtonID(Integer softButtonID) {
        if (softButtonID != null) {
            store.put(Names.softButtonID, softButtonID);
        } else {
        	store.remove(Names.softButtonID);
        }
    }
    public Integer getSoftButtonID() {
        return (Integer) store.get(Names.softButtonID);
    }
    public void setSystemAction(SystemAction systemAction) {
        if (systemAction != null) {
            store.put(Names.systemAction, systemAction);
        } else {
        	store.remove(Names.systemAction);
        }
    }
    public SystemAction getSystemAction() {
    	Object obj = store.get(Names.systemAction);
        if (obj instanceof SystemAction) {
            return (SystemAction) obj;
        } else if (obj instanceof String) {
        	SystemAction theCode = null;
            try {
                theCode = SystemAction.valueForString((String) obj);
            } catch (Exception e) {
            	DebugTool.logError("Failed to parse " + getClass().getSimpleName() + "." + Names.systemAction, e);
            }
            return theCode;
        }
        return null;
    }
}
